server/internal/core/application/covenant.go

```diff
@@ -50,7 +52,7 @@ func NewCovenantService(
 	builder ports.TxBuilder, scanner ports.BlockchainScanner,
 	scheduler ports.SchedulerService,
 ) (Service, error) {
-	eventsCh := make(chan domain.RoundEvent)
+	eventsCh := make(chan interface{})
```

Comment:
🤔

Reply:
I know it looks weird, but if you check how we are using this channel, you will see that we are not using the interface at all. The idea behind this is to extend the event type without modifying the `domain`. 
The new musig2 events are not `domain.RoundEvent` because **they are not modifying the Round state**, and the interface is only used as the parent type to store an array of events in `domain.Round`.

This eventsCh is systematically followed by a type checking `switch` on each event structs. The idea was to make possible to send any type of events (not only domain ones).

Reply:
https://github.com/ark-network/ark/pull/271/files#r1734483986